airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -140,24 +142,97 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
                 "gcp_conn_id": self.gcp_conn_id,
                 "impersonation_chain": self.impersonation_chain,
                 "polling_interval_seconds": self.polling_interval_seconds,
+                "delete_on_error": self.delete_on_error,
             },
         )
 
     async def run(self) -> AsyncIterator[TriggerEvent]:
-        while True:
-            cluster = await self.get_async_hook().get_cluster(
-                project_id=self.project_id, region=self.region, cluster_name=self.cluster_name
+        """Run the trigger."""
+        try:
+            while True:
+                cluster = await self.fetch_cluster_status()
+                if self.check_cluster_state(cluster.status.state):
+                    if cluster.status.state == ClusterStatus.State.ERROR:
+                        await self.gather_diagnostics_and_maybe_delete(cluster)
+                    else:
+                        yield TriggerEvent(
+                            {
+                                "cluster_name": self.cluster_name,
+                                "cluster_state": cluster.status.state,
+                                "cluster": cluster,
+                            }
+                        )
+                    break
+                self.log.info("Sleeping for %s seconds.", self.polling_interval_seconds)
+                await asyncio.sleep(self.polling_interval_seconds)
+        except asyncio.CancelledError:
+            await self.handle_cancellation()
+
+    async def fetch_cluster_status(self) -> Cluster:
```

Comment:
Looks like we're fetching cluster instead of cluster_status. Or is there anything I missed?

Reply:
renamed